File: python/26. Remove Duplicates from Sorted Array.py
class Solution(object):
    def removeDuplicates(self, nums):
        """
        :type nums: List[int]
        :rtype: int
        """
        i = j = k = 0
        ln = len(nums)
        while i < ln:
            nums[k] = nums[i]
            k += 1
            j = i + 1
            while j < ln and nums[i] == nums[j]:
                j += 1
            i = j
        return k
        

File: go/src/leetcode/26. Remove Duplicates from Sorted Array/26. Remove Duplicates from Sorted Array.go
package main

import "fmt"

func removeDuplicates(nums []int) int {
	k := 0
	for i := 1; i < len(nums); i++ {
		if nums[i] != nums[i-1] {
			k++
			nums[k] = nums[i]
		}
	}
	return k + 1
}

func main() {
	fmt.Println(removeDuplicates([]int{1,1,2}))
	fmt.Println(removeDuplicates([]int{0,0,1,1,1,2,2,3,3,4}))
}